webhooks-extension/test/util.sh

```diff
@@ -0,0 +1,215 @@
+        echo "Missing [version]"
+        exit 1
+    fi
+    version="$1"
+    kubectl apply --filename https://github.com/knative/eventing-sources/releases/download/${version}/eventing-sources.yaml
+    # Wait until all the pods come up
+    wait_for_ready_pods knative-sources 180 20
+}
+
+# Install Tekton (instructions here: https://github.com/tektoncd/pipeline/blob/master/docs/install.md#adding-the-tekton-pipelines)
+function install_tekton() {
+    if [ -z "$1" ]; then
+        echo "Usage ERROR for function: install_tekton [version]"
+        echo "Missing [version]"
+        exit 1
+    fi
+    version="$1"
+    kubectl apply --filename https://storage.googleapis.com/tekton-releases/previous/${version}/release.yaml
+    # Wait until all the pods come up
+    wait_for_ready_pods tekton-pipelines 180 20
+}
+
+# Install Dashboard (https://github.com/tektoncd/dashboard)
+#
+# As of May 2nd this is a bit fraught since dashboard does not yet have images pushed to gcr.io
+# 
+# So: look to see if $GOPATH/src/github.com/tektoncd/dashboard exists
+# If it doesn't : check that the user is ok for us to clone and ko apply it
+# If it does: check that the user is ok for us to cd into and ko apply it
+
+function install_dashboard() { 
+  if [ -z "$1" ]; then
+        echo "Usage ERROR for function: install_dashboard [namespace]"
+        echo "Missing [namespace]"
+        exit 1
+  fi
+  namespace="$1"
+  check GOPATH
+  check KO_DOCKER_REPO docker.io/your_docker_id
+
+  dashboard_dir=$GOPATH/src/github.com/tektoncd/dashboard 
+  if [ -d $dashboard_dir ]; then 
+    echo "Dashboard source detected at $dashboard_dir"
+    read -p "Do you wish to ko apply the current source? " -n 1 -r apply
+  else
+    echo "Dashboard source not detected in $dashboard_dir"
+    read -p "Do you wish to git clone and ko apply the master version? " -n 1 -r cloneAndApply
+  fi
+  if [[ $cloneAndApply =~ ^[Yy]$ ]]; then
+    pushd $GOPATH/src/github.com/tektoncd/
+    git clone https://github.com/tektoncd/dashboard.git
+    popd
+    apply="y"
+  fi
+  if [[ $apply =~ ^[Yy]$ ]]; then
+    pushd $GOPATH/src/github.com/tektoncd/dashboard
+    docker login
+    npm install
+    npm run build_ko
+    dep ensure -v
+    ko apply -f config -n $namespace
+    popd
+  fi
+}
+
+# Install https://github.com/tektoncd/experimental/tree/master/webhooks-extension
+function install_webhooks_extension() { 
+  pushd $GOPATH/src/github.com/tektoncd/experimental/webhooks-extension
+  if [ -z "$1" ]; then
+    echo "Usage ERROR for function: install_webhooks_extension [target-namespace]"
+    echo "Missing [namespace]"
+    exit 1
+  fi 
+  namespace=$1
+  docker login
+  dep ensure -v
+  ko apply -f config -n $namespace
+  popd
+  wait_for_ready_pods $namespace 60 10
+}
+
+# Wait until all pods in a namespace are Running or Complete
+# wait_for_ready [namespace] [timeout] <sleepTime>
+# <sleepTime> is optional
+function wait_for_ready_pods() {
```

Comment:
Would something like this be sufficient? https://github.com/openshift-cloud-functions/knative-operators/blob/master/etc/scripts/installation-functions.sh#L32
To wait until there are no pods that are NOT running/completed/terminating? Seems like a big function, but no real qualms otherwise.

Reply:
Nice - thank you, I've added that in.

Reply:
@vtereso thanks for your review, I've made the changes I think necessary so please lgtm if you're happy, thank you.